### ISSUE TITLE
Don't require a `@return` tag for `@interface`

### ISSUE THREAD

**Issue by tschaub** [eslint/eslint#4860]
There is already a condition that disables the check for `@return` for `@class` and `@constructor`.  The [`@interface` tag](http://usejsdoc.org/tags-interface.html) may be applied to a constructor instead of one of these.

With ESLint @ 009a949ccbcd3c7ed9964a4583d7814a398ecfd4, the code below doesn't pass the `valid-jsdoc` rule:
```js
/* eslint valid-jsdoc: 2 */

/**
 * Creates a new thing.
 * @interface
 */
function Thing() {}
```
>   3:1  error  Missing JSDoc @returns for function  valid-jsdoc

If `@interface` is changed to `@constructor`, `valid-jsdoc` passes (as it should).

**Comment by eslintbot** [comment]
@tschaub Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by nzakas** [comment]
Yeah, that looks like a bug.
